src/main/app/common/allowances/allowance.ts

```diff
@@ -24,22 +26,49 @@ export enum LivingAllowanceType {
   DEFENDANT_OVER_25_PARTNER_UNDER_18 = 'DEFENDANT_OVER_25_PARTNER_UNDER_18'
 }
 
+export class Allowances {
+  constructor (public allowance?: Allowance[]) {
+  }
+
+  deserialize (input?: any): Allowance {
+    if (!input) {
+      return input
+    }
+    this.allowance = this.deserializeRows(input.allowances)
+    return this.allowance
+      .sort((a, b) => {
```

Comment:
Can be something like...
`.sort((a, b) => a.startDate.diff(b.startDate, 'days'))`
... for readability